yoonit-camera/src/main/java/ai/cyberlabs/yoonit/camera/analyzers/face/FaceAnalyzer.kt

```diff
@@ -87,73 +88,43 @@ class FaceAnalyzer(
 
                 // Get closest face.
                 // Can be null if no face found.
+                // Used to crop the face image.
                 val closestFace: Face? = this.faceBoundingBoxController.getClosestFace(faces)
 
                 // Transform the camera face coordinates to UI graphic coordinates.
+                // Used to draw the face detection box.
                 val detectionBox = this.faceBoundingBoxController.getDetectionBox(
                     closestFace,
                     image
                 )
 
-                // Get error if exist.
-                val error = this.faceBoundingBoxController.getError(
-                    closestFace,
-                    detectionBox
-                )
-
-                // Emit once if has error.
-                if (error != null) {
-                    if (this.isValid) {
-                        this.isValid = false
-                        this.graphicView.clear()
-                        if (this.cameraEventListener != null) {
-                            if (error != "") {
-                                this.cameraEventListener.onMessage(error)
-                            }
-                            this.cameraEventListener.onFaceUndetected()
-                        }
-                    }
+                // Verify if has error on the closestFace and detectionBox.
+                if (this.hasError(closestFace, detectionBox)) {
                     return@addOnSuccessListener
                 }
-                this.isValid = true
 
-                //Transform media image in bitmap and crop bitmap in the face
-                var faceBoundingBox = Rect(0,0,0,0)
-                closestFace?.let {
-                    faceBoundingBox = it.boundingBox
-                }
-
-                val faceBitmap = when (captureOptions.colorEncoding) {
-                    "YUV" -> cropFaceBitmap(mediaImage.toYUVBitmap(), cameraRotation.toFloat(), faceBoundingBox)
-                    else -> cropFaceBitmap(mediaImage.toRGBBitmap(context), cameraRotation.toFloat(), faceBoundingBox)
-                }
+                // Get face bitmap.
```

Comment:
The let call to instantiate faceBitmap was to avoid using !! at closestFace because normally it isn't a good practice